textract-lambda-cdk-dotnet/Cdk/src/Cdk/CdkStack.cs

```diff
@@ -0,0 +1,81 @@
+using Amazon.CDK;
+using Amazon.CDK.AWS.IAM;
+using Amazon.CDK.AWS.Lambda;
+using Amazon.CDK.AWS.Lambda.EventSources;
+using Amazon.CDK.AWS.Logs;
+using Amazon.CDK.AWS.S3;
+using Constructs;
+
+namespace Cdk
+{
+    public class CdkStack : Stack
+    {
+        readonly string lambdaFunctionName = "textractLambdaFunction";
+        internal CdkStack(Construct scope, string id, IStackProps props = null) : base(scope, id, props)
+        {
+            //Create S3 bucket
+            var bucket = new Bucket(this, "textract-bucket", new BucketProps
+            {
+                BucketName = $"{this.Account}-textract-bucket"
+            });
+
+            //Lambda Function IAM role
+            var lambdaIAMRole = new Role(this, "TextractLambdaRole", new RoleProps
+            {
+                AssumedBy = new ServicePrincipal("lambda.amazonaws.com"),
+                Description = "IAM Role for the lambda function",
+                ManagedPolicies = new[]
+                {
+                        ManagedPolicy.FromAwsManagedPolicyName("service-role/AWSLambdaBasicExecutionRole"),
+                        ManagedPolicy.FromAwsManagedPolicyName("AmazonTextractFullAccess")
```

Comment:
Is `FullAccess` really necessary here? Please try to apply the principle of least privilege wherever reasonably possible

Reply:
Good catch! Done